plugins/org.eclipse.xtend.core/src/org/eclipse/xtend/core/compiler/batch/XtendBatchCompiler.java

```diff
@@ -733,6 +735,15 @@ public boolean apply(String input) {
 		// for annotation processing we need to have the compiler's classpath as a parent.
 		annotationProcessingClassLoader = createClassLoader(classpath, currentClassLoader);
 		resourceSet.eAdapters().add(new ProcessorInstanceForJvmTypeProvider.ProcessorClassloaderAdapter(annotationProcessingClassLoader));
+		removePossiblyShadowedTypeResources(resourceSet);
+	}
+
+	private void removePossiblyShadowedTypeResources(ResourceSet resourceSet) {
```

Comment:
How can you be sure that no existing x-ref was resolved to one of these type resources?